### PR TITLE
[sailfish-browser] Removed remorse timer from delete bookmark. Contri…

### PR DESCRIPTION
**Pull request by antseppa**
…butes to JB#27571

### CODE REVIEW

File: src/pages/components/FavoriteGrid.qml [hunk 1]
```diff
@@ -61,16 +61,8 @@ SilicaGridView {
     delegate: ListItem {
         id: container
 
-        property Item remorse
-
         function remove() {
```
Comment (rainemak) [inline src/pages/components/FavoriteGrid.qml]: I'd also remove this function and move ```bookmarkModel.removeBookmark(url)``` to the FavoriteContextMenu. Matter of taste as currently FavoriteContextMenu is not using ids from elsewhere and that's also good. Not big deal.

For safety, add url parameter for the remove function is you don't remove this. Now url refers to the model.url but it looks quite odd (maybe I have added at some point of time) as call is coming through FavoriteContextMenu. Url is passed to the context menu in the show.

**Comment by antseppa** [comment]
Updated: removed remove() function and moved bookmarkModel.removeBookmark(url) to context menu.

**Comment by antseppa** [comment]
Updated: added a commit which updates indices in bookmarkIndexes map after a bookmark has been removed. Without this, consecutive calls to DeclarativeBookmarkModel::removeBookmark either crashed browser or caused wrong bookmark to be deleted if the index of the bookmark was greater than previous one.

File: src/pages/components/FavoriteContextMenu.qml [hunk 2]
```diff
@@ -50,7 +50,9 @@ Component {
             //: "Remove favorited / bookmarked web page"
             //% "Remove favorite"
             text: qsTrId("sailfish_browser-me-remove_favorite")
-            onClicked: delegate.remove()
+            onClicked: {
+                bookmarkModel.removeBookmark(url)
+            }
```
Comment (rainemak) [inline src/pages/components/FavoriteContextMenu.qml]: No need for brackets.

**Comment by rainemak** [comment]
LGTM

**Pull request merged**